src/main/java/com/spikes2212/command/genericsubsystem/commands/smartmotorcontrollersubsystem/MoveSmartMotorControllerSubsystem.java

```diff
@@ -0,0 +1,153 @@
+        addRequirements(subsystem);
+        this.subsystem = subsystem;
+        this.controlMode = controlMode;
+        this.controlType = controlType;
+        this.pidSettings = pidSettings;
+        this.feedForwardSettings = feedForwardSettings;
+        this.setpoint = setpoint;
+        this.lastTimeNotOnTarget = 0;
+    }
+
+    /**
+     * Constructs a new instance of {@link MoveSmartMotorControllerSubsystem} for a subsystem which consists of
+     * CTRE motor controllers.
+     *
+     * @param subsystem the {@link SmartMotorControllerSubsystem} this command will run on
+     * @param pidSettings the loop's pid constants
+     * @param feedForwardSettings the loop's feed forward gains
+     * @param controlMode the loop's control mode (e.g. voltage, velocity, position...).
+     *                    Only applicable when using CTRE motor controllers
+     * @param setpoint the setpoint this command should bring the {@link SmartMotorControllerSubsystem} to
+     */
+    public MoveSmartMotorControllerSubsystem(SmartMotorControllerSubsystem subsystem, PIDSettings pidSettings,
+                                             FeedForwardSettings feedForwardSettings,
+                                             ControlMode controlMode, Supplier<Double> setpoint) {
+        this(subsystem, pidSettings, feedForwardSettings, controlMode, null, setpoint);
+    }
+
+    /**
+     * Constructs a new instance of {@link MoveSmartMotorControllerSubsystem} for a subsystem which consists of
+     * {@link CANSparkMax} motor controllers.
+     * @param subsystem the {@link SmartMotorControllerSubsystem} this command will run on
+     * @param pidSettings the loop's pid constants
+     * @param feedForwardSettings the loop's feed forward gains
+     * @param controlType the loop's control mode (e.g. voltage, velocity, position...).
+     *                    Only applicable when using {@link CANSparkMax} motor controllers
+     * @param setpoint the setpoint this command should bring the {@link SmartMotorControllerSubsystem} to
+     */
+    MoveSmartMotorControllerSubsystem(SmartMotorControllerSubsystem subsystem, PIDSettings pidSettings,
+                                             FeedForwardSettings feedForwardSettings,
+                                             CANSparkMax.ControlType controlType, Supplier<Double> setpoint) {
+        this(subsystem, pidSettings, feedForwardSettings, null, controlType, setpoint);
+    }
+
+    /**
+     * Configures the subsystem's control loops.
+     */
+    @Override
+    public void initialize() {
+        subsystem.configureLoop(pidSettings, feedForwardSettings);
+    }
+
+    /**
+     * Updates any control loops running on the subsystem.
+     * This method uses the {@code pidSet} method twice: once assuming the subsystem consists of CTRE motor controllers,
+     * and once assuming it consists of {@link CANSparkMax}s. In any given {@link SmartMotorControllerSubsystem},
```

Comment:
don't use the word once like this twice. maybe write and "once again assuming...". this sounds better, at least in my opinion